src/controllers/controllerengine.cpp

```diff
@@ -933,35 +928,22 @@ void ControllerEngine::trigger(QString group, QString name) {
    Input:   Script filename
    Output:  false if the script file has errors or doesn't exist
    -------- ------------------------------------------------------ */
-bool ControllerEngine::evaluate(const QString& scriptName, QList<QString> scriptPaths) {
+bool ControllerEngine::evaluate(const QFileInfo& scriptFile) {
     if (m_pEngine == nullptr) {
         return false;
     }
 
-    QString filename = "";
-    QFile input;
-
-    if (scriptPaths.length() == 0) {
-        // If we aren't given any paths to search, assume that scriptName
-        // contains the full file name
-        filename = scriptName;
-        input.setFileName(filename);
-    } else {
-        for (const QString& scriptPath : scriptPaths) {
-            QDir scriptPathDir(scriptPath);
-            filename = scriptPathDir.absoluteFilePath(scriptName);
-            input.setFileName(filename);
-            if (input.exists())  {
-                qDebug() << "ControllerEngine: Watching JS File:" << filename;
-                m_scriptWatcher.addPath(filename);
-                break;
-            }
-        }
+    if (!scriptFile.exists()) {
```

Comment:
VERIFY_OR_DEBUG_ASSERT? This should be checked before it gets to ControllerEngine, right?

Reply:
Unsure. It wasn't checked before. To make this a debug assert we'll have to add code that filter out files that don't exist to `ControllerPreset`.

Reply:
I think we should do that.

Reply:
On second thought, maybe not. If we did that, the preferences couldn't show that a file is missing.